src/core/datastructures/tfprimitiveset.cpp

```diff
@@ -146,12 +146,24 @@ std::vector<TFPrimitiveData> TFPrimitiveSet::getUnsorted() const {
     return values;
 }
 
-std::pair<std::vector<float>, std::vector<vec4>> TFPrimitiveSet::getVectors() const {
+std::pair<std::vector<double>, std::vector<vec4>> TFPrimitiveSet::getVectors() const {
+    std::pair<std::vector<double>, std::vector<vec4>> result;
+    result.first.reserve(sorted_.size());
+    result.second.reserve(sorted_.size());
+    for (auto& v : sorted_) {
+        result.first.push_back(static_cast<float>(v->getPosition()));
+        result.second.push_back(v->getColor());
+    }
+
+    return result;
+}
+
+std::pair<std::vector<float>, std::vector<vec4>> TFPrimitiveSet::getVectorsf() const {
```

Comment:
Future PR: We could return a data frame or something that allows it to be extended to multiple attributes. Access could be by column name instead, position, color and so on. Perhaps we could even use a data frame for storage in the TF now that I think of it.